packages/amplify_core/android/src/main/kotlin/com/amazonaws/amplify/amplify_core/Core.kt

```diff
@@ -96,18 +97,20 @@ public class Core : FlutterPlugin, ActivityAware, MethodCallHandler {
     }
 
     private fun onConfigure(@NonNull result: Result, @NonNull version: String, @NonNull config: String) {
-        try {
-            Amplify.configure(AmplifyConfiguration.builder(JSONObject(config))
-                    .addPlatform(UserAgent.Platform.FLUTTER, version)
-                    .build(),
-                    context
-            );
-
-
-
-            result.success(true);
-        } catch (e: AmplifyException) {
-            result.error("AmplifyException", e.message, formatAmplifyException(e) )
+        if (!isConfigured) {
```

Comment:
Is there any concern around thread safety, here?

1. Thread 1 sees `isConfigured` is false, so starts configuring
2. Thread 1 is executing the native configuration before `isConfigured = true`
2. Thread 2 starts configuring, `isConfigured` is still false, so it calls `Amplify.configure(...)`, too.

Reply:
@jamesonwilliams I don't think this is a huge concern if customers follow our documentation, and it's not so much a threading issue (I think) as it is an issue with the method call using futures.  We could potentially ameliorate the issue by making the configure method NOT use a future, which would keep the app on a rail because dart itself is single threaded.